core/src/main/scala/sttp/client4/ws/SyncWebSocket.scala

```diff
@@ -0,0 +1,140 @@
+package sttp.client4.ws
+
+import sttp.client4.Identity
+import sttp.model.Headers
+import sttp.ws.{WebSocket, WebSocketClosed, WebSocketFrame}
+
+/** Allows interacting with a web socket. Interactions can happen:
+  *
+  *   - on the frame level, by sending and receiving raw [[WebSocketFrame]] s
+  *   - using the provided `receive*` methods to obtain concatenated data frames, or string/byte payloads, and the
+  *     `send*` method to send string/binary frames.
+  *
+  * The `send*` and `receive*` methods may result in a failed effect, with either one of [[sttp.ws.WebSocketException]]
+  * exceptions, or a backend-specific exception. Specifically, they will fail with [[WebSocketClosed]] if the web socket
+  * is closed.
+  *
+  * See the `either` and `eitherClose` method to lift web socket closed events to the value level.
+  */
+class SyncWebSocket(val delegate: WebSocket[Identity]) {
```

Comment:
Should we move it to the root `client4` package?

The idea is that everything should be available from a single import:

```scala
import sttp.client4._

object WebSocketSynchronous extends App {
  def useWebSocket(ws: SyncWebSocket): Unit = ???

  val backend = DefaultSyncBackend()

  try
    basicRequest
      .get(uri"wss://ws.postman-echo.com/raw")
      .response(ws.sync.asWebSocket(useWebSocket))
      .send(backend)
  finally backend.close()
}
```

Reply:
I think it would rather go to the sttp-shared project, next to `WebSocket`: https://github.com/softwaremill/sttp-shared/blob/master/ws/src/main/scala/sttp/ws/WebSocket.scala

(it's shared among sttp client / tapir)